libraries/HTTPClient/src/HTTPClient.cpp

```diff
@@ -618,8 +618,13 @@ int HTTPClient::sendRequest(const char * type, uint8_t * payload, size_t size)
         }
 
         // send Payload if needed
-        if(payload && size > 0) {
-            if(_client->write(&payload[0], size) != size) {
+        if(payload && (size > 0)) {
+            size_t totalSent = 0;
+            while(_client->connected() && (totalSent < size))
+            {
+                totalSent += _client->write(&payload[totalSent], (size - totalSent));
```

Comment:
if `write` returns `0` and client is still connected, that would mean that the socket is not ready to be written to. I suggest you catch this situation and delay a bit before retrying. Currently the code will try in a busy loop and on single core chips (or when code runs on the same core and WiFi) this is not a good idea.

Reply:
Something like this?
```suggestion
                const size_t thisSend = _client->write(&payload[totalSent], (size - totalSent));
                totalSent += thisSend;
                if(thisSend == 0) {
                    delay(10);
                }
```

Should we add a delay? Or just bail out if `write` returns 0? I can't remember, but maybe I thought `client->write` was blocking?